nmdc_server/ingest/pipeline.py

```diff
@@ -78,26 +83,49 @@ def load_mags(db: Session, obj: Dict[str, Any], **kwargs) -> LoadObjectReturn:
 
 
 def load_mp_analysis(db: Session, obj: Dict[str, Any], **kwargs) -> LoadObjectReturn:
-    peptides = obj.pop("has_peptide_quantifications", [])
     pipeline = cast(models.MetaproteomicAnalysis, load_mp_analysis_base(db, obj, **kwargs))
 
-    for p in peptides:
-        if not db.query(models.MGAGeneFunction).filter_by(subject=p["best_protein"]).first():
-            continue
+    annotations: Collection = kwargs["annotations"]
+
+    query = annotations.find(
+        {
+            "metaproteomic_analysis_id": pipeline.id,
```

Comment:
@naglepuff I believe we want to add a line here to only get ones where best_protien is set to true:
```
"best_protein": True,
```

Reply:
You're right. Added in c23c4de